### PR TITLE
feat: add methods to get reports - spender

### PR DESCRIPTION
**Pull request by bistaastha**

### CODE REVIEW

File: src/app/core/services/platform/v1/approver/reports.service.ts [hunk 1]
```diff
@@ -1,12 +1,55 @@
...
+import { ReportsQueryParams } from 'src/app/core/models/platform/v1/reports-query-params.model';
+import { PAGINATION_SIZE } from 'src/app/constants';
+import { Report } from 'src/app/core/models/platform/v1/report.model';
 
 @Injectable({
   providedIn: 'root',
 })
 export class ApproverReportsService {
-  constructor(private approverPlatformApiService: ApproverPlatformApiService) {}
+  constructor(
```
Comment (bistaastha) [inline src/app/core/services/platform/v1/approver/reports.service.ts:13]: tests to be added in follow up

**Review by mvaishnavi — APPROVED** [review]

File: src/app/core/services/platform/v1/approver/reports.service.ts [hunk 2]
```diff
@@ -1,12 +1,73 @@
...
+      data: {
+        query_params: `state=${params.state}`,
+      },
+    };
+    return this.approverPlatformApiService
+      .post<{ data: StatsResponse }>('/reports/stats', queryParams)
+      .pipe(map((res) => res.data));
+  }
+
+  getReport(id: string): Observable<Report> {
```
Comment (Chethan-Fyle) [inline src/app/core/services/platform/v1/approver/reports.service.ts:67]: `getReport` -> `getReportById`

File: src/app/core/services/platform/v1/spender/reports.service.spec.ts [hunk 3]
```diff
@@ -141,6 +141,39 @@ describe('SpenderReportsService', () => {
     });
   });
 
+  it('getReport(): should get a report by id', () => {
+    spyOn(spenderReportsService, 'getReportsByParams').and.returnValue(of(allReportsPaginated1));
+    const queryParams = {
+      id: 'eq.rpvcIMRMyM3A',
+    };
+    spenderReportsService.getReport('rpvcIMRMyM3A').subscribe((res) => {
+      console.log(res);
```
Comment (Chethan-Fyle) [inline src/app/core/services/platform/v1/spender/reports.service.spec.ts:150]: Remove console log.

File: src/app/core/services/platform/v1/spender/reports.service.ts [hunk 4]
```diff
@@ -18,18 +20,26 @@ export class SpenderReportsService {
     private spenderPlatformV1ApiService: SpenderPlatformV1ApiService
   ) {}
 
-  getAllReportsByParams(queryParams: ReportsQueryParams): Observable<Report[]> {
+  getAllReportsByParams(queryParams: ReportsQueryParams, order?: string): Observable<Report[]> {
```
Comment (Chethan-Fyle) [inline src/app/core/services/platform/v1/spender/reports.service.ts:23]: Why not add `order` to `ReportsQueryParams`?

**Pull request merged**